network/networkutils/iptables.go

```diff
@@ -0,0 +1,16 @@
+package networkutils
+
+import "github.com/Azure/azure-container-networking/iptables"
+
+type IPTablesClientInterface interface {
```

Comment:
we [don't declare interfaces like this](https://medium.com/@cep21/preemptive-interface-anti-pattern-in-go-54c18ac0668a) in Go.

TL;DR: Go has implicit interface implementation. If a type has the methods, it implements the interface. You don't need to explicitly declare that it implements the interface like in Java/C# or other languages.
Therefore, an interface can be satisfied by any type that implements all of the interface methods in its method set... Or, if we approach it from the other direction, an interface can be defined over a subset of the methods of an object(s).
Interfaces are for the _consumer_, not the producer. The consumer defines the interface that describes only the behavior that it needs, and any object which satisfies that behavioral contract can be passed in as the backing implementation of that interface.